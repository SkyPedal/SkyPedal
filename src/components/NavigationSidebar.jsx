import React from "react";
import { Link } from "react-router-dom";

const NavigationSidebar = () => {
  return (
    <div className="w-1/5 sticky left-0 border-r-4">
      <ul className="p-5">
        <li className="p-5">
          <Link to="/">Home</Link>
        </li>
        <li className="p-5">
          <Link to="/">Record</Link>
        </li>
        <li className="p-5">
          <Link to="/">Compete</Link>
        </li>
        <li className="p-5">
          <Link to="/">Rewards</Link>
        </li>
        <li className="p-5 absolute bottom-0">
          <Link to="/">Sign in</Link>
        </li>
      </ul>
    </div>
  );
};

export default NavigationSidebar;
