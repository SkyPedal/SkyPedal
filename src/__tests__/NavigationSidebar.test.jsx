import { render } from "@testing-library/react";
import { describe, expect, it } from "vitest";
import NavigationSidebar from "../components/NavigationSidebar";
import { BrowserRouter as Router } from "react-router-dom";

describe("NavigationSidebar", () => {
  it("snapshot renders", () => {
    const { asFragment } = render(
      <Router>
        <NavigationSidebar />
      </Router>,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
