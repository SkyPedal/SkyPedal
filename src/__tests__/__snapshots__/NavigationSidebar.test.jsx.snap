// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`NavigationSidebar > snapshot renders 1`] = `
<DocumentFragment>
  <div
    class="w-20 md:w-52 sticky left-0 border-r-4 border-solid"
    style="border-image: var(--sky-gradient-shape) 1;"
  >
    <ul
      class="p-5"
    >
      <li
        class="md:p-5 pt-5 pb-5"
      >
        <a
          href="/"
        >
          <div
            class="flex h-12 items-center"
          >
            <img
              alt="Home"
              class="w-8"
              src="/src/assets/icons/Home.svg"
            />
            <p
              class="pl-4 hidden md:block"
            >
              Home
            </p>
          </div>
        </a>
      </li>
      <li
        class="md:p-5 pt-5 pb-5"
      >
        <a
          href="/"
        >
          <div
            class="flex h-12 items-center"
          >
            <img
              alt="Record"
              class="w-8"
              src="/src/assets/icons/Add.svg"
            />
            <p
              class="pl-4 hidden md:block"
            >
               Record 
            </p>
          </div>
        </a>
      </li>
      <li
        class="md:p-5 pt-5 pb-5"
      >
        <a
          href="/"
        >
          <div
            class="flex h-12 items-center"
          >
            <img
              alt="Compete"
              class="w-8"
              src="/src/assets/icons/Trophy.svg"
            />
            <p
              class="pl-4 hidden md:block"
            >
               Compete 
            </p>
          </div>
        </a>
      </li>
      <li
        class="md:p-5 pt-5 pb-5"
      >
        <a
          href="/"
        >
          <div
            class="flex h-12 items-center"
          >
            <img
              alt="Rewards"
              class="w-8"
              src="/src/assets/icons/Rewards.svg"
            />
            <p
              class="pl-4 hidden md:block"
            >
               Rewards 
            </p>
          </div>
        </a>
      </li>
      <li
        class="md:p-5 pt-5 pb-5 absolute bottom-0"
      >
        <a
          href="/"
        >
          <div
            class="flex h-12 items-center"
          >
            <img
              alt="SignIn"
              class="w-8"
              src="/src/assets/icons/SignIn.svg"
            />
            <p
              class="pl-4 hidden md:block"
            >
               Sign In
            </p>
          </div>
        </a>
      </li>
    </ul>
  </div>
</DocumentFragment>
`;
